src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -176,6 +176,15 @@ public class CoinstakeContext
             public Key Key { get; set; }
         }
 
+        /// <summary>
+        /// Indicates the current state of the staking: in progress, stopped or idle.
```

Comment:
This is weird, the comment mentions three states, but the enumeration only has two states

Reply:
I personally think you should have three or four states here and only have one field, not two. The way you have it now will not work, just consider what would happen if you executed start – stop – start – stop – start

Reply:
Fixed comment.

Not sure how we can get away with just one flag. Every time we run Interlocked.CompareExchange it sets the flag to one of the states, i.e. I cannot just check the state without setting the flag. Unless I am missing something.

Reply:
I think you are missing something. But let's start with a simple question – do you see why it cannot work as it is?